src/mongocxx/options/index.hpp

```diff
@@ -0,0 +1,361 @@
+        /// @return The current config_str.
+        ///
+        const stdx::optional<std::string>& config_str() const;
+
+       private:
+        friend collection;
+        MONGOCXX_PRIVATE const int type() const override;
+        stdx::optional<std::string> _config_str;
+    };
+
+    index();
+
+    ///
+    /// Whether or not to build the index in the background so that building
+    /// the index does not block other database activities.
+    ///
+    /// @param background
+    ///   Whether or not to build the index in the background.
+    ///
+    /// @see https://docs.mongodb.org/manual/tutorial/build-indexes-in-the-background/
+    ///
+    void background(bool background);
+
+    ///
+    /// The current background setting.
+    ///
+    /// @return The current background.
+    ///
+    const stdx::optional<bool>& background() const;
+
+    ///
+    /// Whether or not to create a unique index so that the collection will not
+    /// accept insertion of documents where the index key or keys match an
+    /// existing value in the index.
+    ///
+    /// @param unique
+    ///   Whether or not to create a unique index.
+    ///
+    void unique(bool unique);
+
+    ///
+    /// The current unique setting.
+    ///
+    /// @return The current unique.
+    ///
+    /// @see https://docs.mongodb.org/manual/core/index-unique/
+    ///
+    const stdx::optional<bool>& unique() const;
+
+    ///
+    /// The name of the index.
+    ///
+    /// @param name
+    ///   The name of the index.
+    ///
+    void name(std::string name);
+
+    ///
+    /// The current name setting.
+    ///
+    /// @return The current name.
+    ///
+    const stdx::optional<std::string>& name() const;
+
+    ///
+    /// Whether or not to create a sparse index. Sparse indexes only reference
+    /// documents with the indexed fields.
+    ///
+    /// @param sparse
+    ///   Whether or not to create a sparse index.
+    ///
+    void sparse(bool sparse);
+
+    ///
+    /// The current sparse setting.
+    ///
+    /// @return The current sparse.
+    ///
+    const stdx::optional<bool>& sparse() const;
+
+    ///
+    /// Optionally used only in MongoDB 3.0.0 and higher. Specifies the storage
+    /// engine options for the index
+    ///
+    /// @warning
+    ///  The base_storage_options* instance must stay in scope until after
+    ///  the call to create_index.
+    ///
+    /// @param storage_options
+    ///   The storage engine options for the index in.
+    ///
+    void storage_options(base_storage_options* storage_options);
+
+    ///
+    /// The current storage_options setting.
+    ///
+    /// @return The current storage_options.
+    ///
+    const base_storage_options* storage_options() const;
+
+    ///
+    /// Set a value, in seconds, as a TTL to control how long MongoDB retains
+    /// documents in this collection.
+    ///
+    /// @param expire_after_seconds
+    ///   The amount of time, in seconds, to retain documents.
```

Comment:
Thanks for the doc's comments :) Should have been more careful when putting links...

Fixed!